common/lib/dependabot/utils.rb

```diff
@@ -44,5 +44,15 @@ def self.always_clone_for_package_manager?(package_manager)
     def self.register_always_clone(package_manager)
       @cloning_package_managers << package_manager
     end
+
+    def self.wildcard_match?(wildcard_string, candidate_string)
+      return false unless wildcard_string && candidate_string
+
+      regex_string = "a#{wildcard_string.downcase}a".split("*").
```

Comment:
Question: what's up with these `a` characters? Is there a test case that fails without them?

Reply:
Yeah looks like there's a few test cases for this fail, e.g. `wildcard_match?("*", "bus")`, looks like we just add a random character so we end up with multiple parts when splitting by `*` so we can join `.*` on the parts